acme/acme/client.py

```diff
@@ -358,14 +358,18 @@ def poll_and_request_issuance(
 
         # priority queue with datetime.datetime (based on Retry-After) as key,
         # and original Authorization Resource as value
-        waiting = [(datetime.datetime.now(), authzr) for authzr in authzrs]
+        waiting = [
+            (datetime.datetime.now(), index, authzrs[index])
+            for index in range(len(authzrs))
```

Comment:
Might this be clearer with `enumerate`?

`[(datetime.datetime.now(), index, authrz) for index, authrz in enumerate(authzrs)]`